### PR TITLE
Cleaning up project.json

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

File: src/dotnet-test-xunit/project.json [hunk 1]
```diff
@@ -25,14 +25,13 @@
   },
   "dependencies": {
     "NETStandard.Library": "1.0.0-rc2-23901",
-    "System.Threading.ThreadPool": "4.0.10-rc2-23901",
     "Microsoft.Extensions.Testing.Abstractions": "1.0.0-beta-001636",
     "Microsoft.DotNet.ProjectModel": "1.0.0-beta-001636",
     "Microsoft.DotNet.Cli.Utils": "1.0.0-beta-001636",
-    "Microsoft.Extensions.CommandLineUtils.Sources": "1.0.0-*",
-    "NuGet.Packaging.Core": "3.4.0-beta-625",
+    "Microsoft.Extensions.CommandLineUtils.Sources": { "version": "1.0.0-*", "type": "build" },
```
Comment (pranavkm) [inline src/dotnet-test-xunit/project.json:31]: Making sources package type: build

File: src/dotnet-test-xunit/project.json [hunk 2]
```diff
@@ -25,14 +25,13 @@
   },
   "dependencies": {
     "NETStandard.Library": "1.0.0-rc2-23901",
-    "System.Threading.ThreadPool": "4.0.10-rc2-23901",
     "Microsoft.Extensions.Testing.Abstractions": "1.0.0-beta-001636",
     "Microsoft.DotNet.ProjectModel": "1.0.0-beta-001636",
     "Microsoft.DotNet.Cli.Utils": "1.0.0-beta-001636",
-    "Microsoft.Extensions.CommandLineUtils.Sources": "1.0.0-*",
-    "NuGet.Packaging.Core": "3.4.0-beta-625",
```
Comment (pranavkm) [inline src/dotnet-test-xunit/project.json:33]: Removing transitive dependencies.

**Comment by pranavkm** [comment]
I could send the second commit as a different PR. But apparently more stuff in the test discovery needs a try-catch.

File: src/dotnet-test-xunit/project.json (hunk 2 above)
Comment (livarcocc) [inline src/dotnet-test-xunit/project.json:33]: We don't need NuGet anymore?

Comment (pranavkm) [inline src/dotnet-test-xunit/project.json:33]: It gets pulled in transitively from `DotNet.ProjectModel`. The problem right now is that they are referencing a different version (3.4.0-rtm-*) which results in downgrade warnings.

Comment (livarcocc) [inline src/dotnet-test-xunit/project.json:33]: Well, if we have a dependency on it we shouldn't rely on transitivity to pull it in. No?

Comment (pranavkm) [inline src/dotnet-test-xunit/project.json:33]: We actually don't reference NuGet in this code.

**Comment by livarcocc** [comment]
:shipit:

**Comment by pranavkm** [comment]
Could you merge this in? I don't have push permissions to the xunit repo

**Comment by pranavkm** [comment]
Thanks!

**Pull request merged**